### PR TITLE
[ios only] prevent logging of password in nightly builds

### LINKED ISSUE #6430: [ios] [nightly builds] password is printed in the logs when "Save password" option is used
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: password is printed in the logs when "Save password" option is used

#### Expected behavior
no password in the logs

#### Actual behavior
password in the logs

### Reproduction

1. on iOS create account
2. log out
3. on Sign in screen turn ON toggle to Save password until login
4. type password and login
5. close app
6. launch app and wait till app will sign in automatically
7. check console log for the password

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

- [ ] wrap password with `security/mask-data`
- [ ] stop printing of re-frame events' parameters

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: nightly
* Operating System: iOS

#### Logs

[comment]: # (if on Android please replicate bug whilst running adb logcat)
```
default	16:42:31.648572 +0300	StatusIm	DEBUG [status-im.utils.handlers:36] - Handling re-frame event:  :accounts.login.callback/get-user-password-success mypassword
```

### PR DESCRIPTION
**Pull request by rasom**
fix #6430 

status: ready

### CODE REVIEW

**Review by mandrigin — APPROVED** [review]

**Review by dmitryn — APPROVED** [review]

**Comment by annadanchenko** [comment]
@rasom app is crashing when signing in with saved password. 
Also as you can see in Console password "bolcom" is printed in new place. 
[geth.log](https://github.com/status-im/status-react/files/2498929/geth.log)

```
default	10:56:59.508835 +0300	StatusIm	Exception '-[__NSDictionaryM UTF8String]: unrecognized selector sent to instance 0x1c4230dc0' was thrown while invoking login on target Status with params (
    9aa4d4ea0236295173b0e79c271385d495a341f4,
        {
        data = bolcom;
    },
    281
)
callstack: (
	0   CoreFoundation                      0x000000018186bd50 <redacted> + 148
	1   libobjc.A.dylib                     0x0000000180d80528 objc_exception_throw + 56
	2   CoreFoundation                      0x00000001818791f8 <redacted> + 0
	3   CoreFoundation                      0x00000001818716e4 <redacted> + 1380
	4   CoreFoundation                      0x00000001817570dc _CF_forwarding_prep_0 + 92
	5   StatusIm                            0x0000000100f5e124 _ZN5folly6detail15str_to_integralIyEENS_8ExpectedIT_NS_14ConversionCodeEEEPNS_5RangeIPKcEE + 11703632
	6   CoreFoundation                      0x00000001818736a0 <redacted> + 144
	7   CoreFoundation                      0x0000000181752820 <redacted> + 292
	8   CoreFoundation           <…>
default	10:56:59.508943 +0300	StatusIm	*** Terminating app due to uncaught exception 'RCTFatalException: Exception '-[__NSDictionaryM UTF8String]: unrecognized selector sent to instance 0x1c4230dc0' was thrown while invoking login on target Status with params (
    9aa4d4ea0236295173b0e79c271385d495a341f4,
        {
        data = bolcom;
    },
    281
)
callstack: (
	0   CoreFoundation                      0x000000018186bd50 <redacted> + 148
	1   libobjc.A.dylib                     0x0000000180d80528 objc_exception_throw + 56
	2   CoreFoundation                      0x00000001818791f8 <redacted> + 0
	3   CoreFoundation                      0x00000001818716e4 <redacted> + 1380
	4   CoreFoundation                      0x00000001817570dc _CF_forwarding_prep_0 + 92
	5   StatusIm                            0x0000000100f5e124 _ZN5folly6detail15str_to_integralIyEENS_8ExpectedIT_NS_14ConversionCodeEEEPNS_5RangeIPKcEE + 11703632
	6   CoreFoundation                      0x00000001818736a0 <redacted> + 144
	7   CoreFoundation                      <…>

```
 
```
defalt	10:56:59.461792 +0300	StatusIm	StartNode result {"error":""}
debug	10:56:59.484651 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.484699 +0300	StatusIm	sa_dst_compare_internal 82.193.117.90:0@0 = 193.108.240.3:0@0
debug	10:56:59.484729 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.484750 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d2, known while d1 not known
debug	10:56:59.484796 +0300	StatusIm	sa_dst_compare_internal 193.27.209.211:0@0 < 82.193.117.90:0@0
debug	10:56:59.484828 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.485077 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d2, known while d1 not known
debug	10:56:59.485127 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.485187 +0300	StatusIm	sa_dst_compare_internal 91.236.251.5:0@0 < 193.27.209.211:0@0
debug	10:56:59.485605 +0300	StatusIm	sa_dst_compare_internal 91.236.251.30:0@0 = 91.198.10.1:0@0
debug	10:56:59.485636 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.485669 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d2, known while d1 not known
debug	10:56:59.485719 +0300	StatusIm	sa_dst_compare_internal 195.78.244.34:0@0 < 91.236.251.30:0@0
debug	10:56:59.485761 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.485802 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d2, known while d1 not known
debug	10:56:59.485852 +0300	StatusIm	sa_dst_compare_internal 212.110.158.29:0@0 < 195.78.244.34:0@0
debug	10:56:59.493134 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.493208 +0300	StatusIm	sa_dst_compare_internal 194.40.240.24:0@0 = 212.110.158.28:0@0
debug	10:56:59.493298 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.494189 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d2, known while d1 not known
debug	10:56:59.494234 +0300	StatusIm	sa_dst_compare_internal 194.40.240.12:0@0 < 194.40.240.24:0@0
debug	10:56:59.494500 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.494545 +0300	StatusIm	sa_dst_compare_internal 82.193.117.90:0@0 = 194.40.240.12:0@0
debug	10:56:59.500383 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.500436 +0300	StatusIm	sa_dst_compare_internal 91.236.251.30:0@0 > 2001:67c:440:400::5.0@0
debug	10:56:59.500472 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.500546 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d2, known while d1 not known
debug	10:56:59.500595 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 < 91.236.251.30:0@0
debug	10:56:59.500638 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
default	10:56:59.500695 +0300	StatusIm	DEBUG [status-im.utils.handlers:34] - Handling re-frame event:  :signals/signal-received
debug	10:56:59.500809 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 2001:67c:440:400::5.0@0
debug	10:56:59.500869 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.500918 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 2001:67c:440:824:91:236:251:24.0@0
debug	10:56:59.500953 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.501026 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 2001:67c:440:400::5.0@0
debug	10:56:59.501078 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.501124 +0300	StatusIm	sa_dst_compare_internal_wrapper prefer d1, known while d2 not known
debug	10:56:59.501151 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 194.40.240.30:0@0
debug	10:56:59.501205 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.501251 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 = 91.198.10.1:0@0
debug	10:56:59.501298 +0300	StatusIm	sa_dst_compare_statistics no preference, equal min rtt 14.750000
debug	10:56:59.501332 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 = 91.236.251.30:0@0
default	10:56:59.501385 +0300	StatusIm	DEBUG [status-im.events:647] - :event-str {"type":"discovery.started","event":null}
debug	10:56:59.501433 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.501486 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 2001:67c:2e74:1::17.0@0
debug	10:56:59.501533 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.501583 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 2a01:6d80::195:20:96:8.0@0
debug	10:56:59.501628 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.501692 +0300	StatusIm	sa_dst_compare_internal 62.149.0.30:0@0 > 2001:67c:440:824:91:236:251:24.0@0
debug	10:56:59.501722 +0300	StatusIm	sa_dst_compare_statistics prefer 1, dst 2 has no route
debug	10:56:59.501756 +0300	StatusIm	sa_dst_compare_internal 194.40.240.30:0@0 > 2001:67c:2e74:1::17.0@0
debug	10:56:59.501806 +0300	StatusIm	sa_dst_compare_internal 2001:67c:440:400::5.0@0 = 2001:67c:2e74:1::17.0@0
debug	10:56:59.501853 +0300	StatusIm	sa_dst_compare_internal 2001:67c:440:824:91:236:251:24.0@0 = 2001:67c:440:400::5.0@0
debug	10:56:59.501894 +0300	StatusIm	sa_dst_compare_internal 2a01:6d80::195:20:96:8.0@0 = 2001:67c:440:824:91:236:251:24.0@0
default	10:56:59.501936 +0300	StatusIm	DEBUG [status-im.signals.core:64] - Event  discovery.started  not handled
default	10:56:59.502086 +0300	StatusIm	DEBUG [status-im.utils.handlers:34] - Handling re-frame event:  :signals/signal-received
default	10:56:59.502163 +0300	StatusIm	DEBUG [status-im.events:647] - :event-str {"type":"node.started","event":null}
default	10:56:59.503021 +0300	StatusIm	DEBUG [status-im.signals.core:64] - Event  node.started  not handled
default	10:56:59.504570 +0300	StatusIm	DEBUG [status-im.utils.handlers:34] - Handling re-frame event:  :signals/signal-received
default	10:56:59.505694 +0300	StatusIm	DEBUG [status-im.events:647] - :event-str {"type":"node.ready","event":null}
default	10:56:59.506870 +0300	StatusIm	-[__NSDictionaryM UTF8String]: unrecognized selector sent to instance 0x1c4230dc0
default	10:56:59.508835 +0300	StatusIm	Exception '-[__NSDictionaryM UTF8String]: unrecognized selector sent to instance 0x1c4230dc0' was thrown while invoking login on target Status with params (
    9aa4d4ea0236295173b0e79c271385d495a341f4,
        {
        data = bolcom;
    },
    281
)
callstack: (
	0   CoreFoundation                      0x000000018186bd50 <redacted> + 148
	1   libobjc.A.dylib                     0x0000000180d80528 objc_exception_throw + 56
	2   CoreFoundation                      0x00000001818791f8 <redacted> + 0
	3   CoreFoundation                      0x00000001818716e4 <redacted> + 1380
	4   CoreFoundation                      0x00000001817570dc _CF_forwarding_prep_0 + 92
	5   StatusIm                            0x0000000100f5e124 _ZN5folly6detail15str_to_integralIyEENS_8ExpectedIT_NS_14ConversionCodeEEEPNS_5RangeIPKcEE + 11703632
	6   CoreFoundation                      0x00000001818736a0 <redacted> + 144
	7   CoreFoundation                      0x0000000181752820 <redacted> + 292
	8   CoreFoundation           <…>
default	10:56:59.508943 +0300	StatusIm	*** Terminating app due to uncaught exception 'RCTFatalException: Exception '-[__NSDictionaryM UTF8String]: unrecognized selector sent to instance 0x1c4230dc0' was thrown while invoking login on target Status with params (
    9aa4d4ea0236295173b0e79c271385d495a341f4,
        {
        data = bolcom;
    },
    281
)
callstack: (
	0   CoreFoundation                      0x000000018186bd50 <redacted> + 148
	1   libobjc.A.dylib                     0x0000000180d80528 objc_exception_throw + 56
	2   CoreFoundation                      0x00000001818791f8 <redacted> + 0
	3   CoreFoundation                      0x00000001818716e4 <redacted> + 1380
	4   CoreFoundation                      0x00000001817570dc _CF_forwarding_prep_0 + 92
	5   StatusIm                            0x0000000100f5e124 _ZN5folly6detail15str_to_integralIyEENS_8ExpectedIT_NS_14ConversionCodeEEEPNS_5RangeIPKcEE + 11703632
	6   CoreFoundation                      0x00000001818736a0 <redacted> + 144
	7   CoreFoundation                      <…>
```

File: src/status_im/utils/security.cljs [hunk 1]
```diff
@@ -17,3 +18,8 @@
 ;; Returns a MaskedData instance that stores the piece of data.
 (defn mask-data [data]
   (MaskedData. data))
+
+(defn unmask-data [data]
```
Comment (mandrigin) [inline src/status_im/utils/security.cljs]: looks unused. either we need to use `unmask-data` in the code, or remove this method

File: src/status_im/utils/security.cljs [hunk 2]
```diff
@@ -1,4 +1,5 @@
-(ns status-im.utils.security)
+(ns status-im.utils.security
+  (:require [taoensso.timbre :as log]))
```
Comment (mandrigin) [inline src/status_im/utils/security.cljs]: looks unnecessary

**Review by mandrigin — CHANGES_REQUESTED** [review]
a few leftovers seems to be there

File: src/status_im/utils/security.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/utils/security.cljs]: https://github.com/status-im/status-react/pull/6431/files#diff-b2c790594a17c6e0263a23b429552b16R103

File: src/status_im/utils/security.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/utils/security.cljs]: fixed

File: src/status_im/utils/security.cljs (hunk 1 above)
Comment (mandrigin) [reply src/status_im/utils/security.cljs]: ah, okay

**Review by mandrigin — APPROVED** [review]

**Comment by annadanchenko** [comment]
@rasom app crashes on entering first char to password field when creating new account. Can't proceed testing as account can't be created:

```
default	12:23:11.812459 +0200	StatusIm	Error: Error: No protocol method Unmaskable.unmask defined for type string: b

This error is located at:
    in create-account
    in reagent15
    in RCTView
    in reagent14
    in r
    in t
    in RCTView
    in n
    in t
    in t
    in n
    in RCTView
    in t
    in RCTView
    in t
    in t
    in RCTView
    in t
    in n
    in n
    in r
    in n
    in t
    in t
    in n
    in n
    in reagent2
    in root
    in reagent1
    in RCTView
    in RCTView
    in t

This error is located at:
    in n
    in t
    in t
    in n
    in n
    in reagent2
    in root
    in reagent1
    in RCTView
    in RCTView
    in t
```

used ipa file from Oct 24 https://ci.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-6431/7/

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-6431/7/">status-react » pull requests » PR-6431 #7 [Jenkins]</a></strong></div></blockquote>

**Pull request merged**